packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js

```diff
@@ -30,14 +31,24 @@ exports.onCreateWebpackConfig = ({ actions, stage }) => {
   })
 }
 
-exports.onPostBuild = async (
-  { store, pathPrefix, reporter },
-  userPluginOptions
-) => {
+exports.onPostBuild = async ({ store, pathPrefix }, userPluginOptions) => {
   const pluginData = makePluginData(store, assetsManifest, pathPrefix)
+
   const pluginOptions = { ...DEFAULT_OPTIONS, ...userPluginOptions }
 
-  const { redirects, pageDataStats, nodes } = store.getState()
+  const { redirects, pageDataStats, nodes, pages } = store.getState()
+
+  /**
+   * Emit via IPC routes for which pages are non SSG
+   */
+  for (const [pathname, page] of pages) {
+    if (page.mode && page.mode !== `SSG`) {
+      emitRoutes({
+        [`${pathname.substring(1)}index.html`]: page.mode,
+        [`page-data/${pathname.substring(1)}page-data.json`]: page.mode,
+      })
```

Comment:
Same as the above

```suggestion
      emitRoute({
        path: `${pathname.substring(1)}index.html`,
        mode: page.mode,
      })
      emitRoute({
        path: `page-data/${pathname.substring(1)}page-data.json`,
        mode: page.mode,
      })
```

Reply:
The way we handle this downstream, I'd rather not do this change and make it easier for us in Cloud

Reply:
but it doesnt matter actually and ill do it

Reply:
actually i dont want to do this to then transform it again. keeping it the same